GLMakie/src/events.jl

```diff
@@ -170,11 +173,12 @@ end
 
 function correct_mouse(screen::Screen, w, h)
     nw = to_native(screen)
-    sf = screen.scalefactor[] / (@static Sys.isapple() ? scale_factor(nw) : 1)
     _, winh = window_size(nw)
-    @static if Sys.isapple()
+    if GLFW.GetPlatform() in (GLFW.PLATFORM_COCOA, GLFW.PLATFORM_WAYLAND)
+        sf = screen.scalefactor[] / scale_factor(nw)
         return w, (winh / sf) - h
     else
+        sf = screen.scalefactor[]
         return w / sf, (winh - h) / sf
     end
```

Comment:
```suggestion
    sf = screen.scalefactor[]
    if GLFW.GetPlatform() in (GLFW.PLATFORM_COCOA, GLFW.PLATFORM_WAYLAND)
        sf /= scale_factor(nw)
    end
    return w / sf, (winh - h) / sf
```
seems to work better for me when I change between scale factors 1, 2, and 3.